File: hw/vizcaino/u3/hw25CompositePattern/hw25CompositePatternJavaScript/clerk.js
const Employee = require('./employee');

class Clerk extends Employee {
    constructor(name) {
        super();
        this.title = "Clerk";
        if (name) {
            this.name = name;
        }
    }

    stateName() {
        super.stateName();
    }
}

module.exports = Clerk;

File: hw/codena/u3/hw24CompositePatterCode/Java/CompositePattern/src/main/java/ec/edu/espe/compositepattern/model/Clerk.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.compositepattern.model;

/**
 *
 * @author Daniel Codena, CodeBreakers, @ESPE
 */
public class Clerk extends Employee {

    public Clerk(String aName) {
        this();
        name = aName;
    }

    public Clerk() {
        title = "Clerk";
    }

    public void stateName() {
        super.stateName();
    }
}
